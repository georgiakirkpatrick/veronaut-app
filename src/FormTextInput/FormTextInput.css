.FormTextInput {
    width: 100%;
    margin-bottom: 2vh;
}

.FormTextInput input {
    display: block;
    width: 100%;
    margin: 5px 0px 10px;
    padding: 10px;
    font: inherit;
    width: 100%;
}