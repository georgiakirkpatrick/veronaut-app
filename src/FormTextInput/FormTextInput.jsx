import React from 'react'
import './FormTextInput.css'

const FormTextInput = ({
  currentValue = 1,
  handleChange = () => {},
  id = 1,
  name = '',
  prompt = ''
}) => {
  return (
    <div className='FormTextInput'>
      <label htmlFor={id}>{prompt}</label>
      
      <input
        required
        id={id}
        name={name}
        onChange={handleChange}
        value={currentValue}
        type='text'
      />
    </div>
  )
}

export default FormTextInput