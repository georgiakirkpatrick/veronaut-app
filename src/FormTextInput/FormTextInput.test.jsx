import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import FormTextInput from './FormTextInput'

describe('FormTextInput', () => {
    it ('renders without crashing', () => {
        render(<FormTextInput />)
        screen.debug()
    })

    it ('render the UI as expected', () => {
        const tree = render(<FormTextInput />)
        expect(tree).toMatchSnapshot()
    })
})