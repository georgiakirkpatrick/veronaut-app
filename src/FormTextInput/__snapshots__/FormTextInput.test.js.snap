// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput render the UI as expected 1`] = `
<div
  className="FormTextInput"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <input
    id={1}
    name=""
    onChange={[Function]}
    required={true}
    type="text"
    value={1}
  />
</div>
`;
